push-to-neofs.py

```diff
@@ -0,0 +1,159 @@
+        type=str,
+        help="Path to the directory with the files to be pushed",
+    ),
+    parser.add_argument(
+        "--expire-at",
+        type=int,
+        help="Expiration epoch. If epoch is not provided, or if it is 0, the report will be stored indefinitely",
+        default=None,
+    )
+
+    return parser.parse_args()
+
+
+def get_password() -> str:
+    password = os.getenv("NEOFS_WALLET_PASSWORD")
+    return password
+
+
+def change_root_dir_to_container_id(
+    # root_directory: str, current_directory: str, run_id: str
+    root_directory: str, filepath: str, run_id: str
+) -> str:
+    """
+    The root of the directory is changing to Container ID.
+    For example, if the directory is /home/varg/work/neofs-testcases/report:
+        varg@burzum:~/work/neofs-testcases/report$ tree -d
+        .
+        ├── data
+        │   ├── attachments
+        │   └── test-cases
+        ├── export
+        ├── history
+    And
+        Container ID is HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6
+        Run ID is 832-1695916423
+    Then the directory structure in the NeoFS will be like this:
+        HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6
+        ├── data
+        │   ├── attachments
+        │   └── test-cases
+        ├── export
+        ├── history
+    And the static page will be available at the link:
+    https://http.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/832-1695916423/data/__files__
+    https://http.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/832-1695916423/data/attachments/__files__
+    ...
+    https://http.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/832-1695916423/export/__files__
+    ...
+
+    This adds the ability to upload a static page (such as the Allure report) and hyperlink from that static page
+    to attachments that are separate objects.
+    In this way we can split a static page from one huge file of several hundred megabytes to the wonderful
+    world of Web 1.0 with hyperlinks.
+    Isn't that cool?
+    """
+    relative_path = os.path.relpath(filepath, root_directory)
+    return os.path.join(run_id, relative_path)
+
+
+def push_files_to_neofs(
+    directory: str,
+    neofs_domain: str,
+    wallet: str,
+    cid: str,
+    run_id: str,
+    expire_at: int,
+    password: str,
+) -> None:
+    if not os.path.exists(directory):
+        raise Exception(f"Directory '{directory}' does not exist.")
+    if not os.listdir(directory):
+        raise Exception(f"Directory '{directory}' is empty.")
+
+    base_cmd = (
+        f"NEOFS_CLI_PASSWORD={password} neofs-cli --rpc-endpoint {neofs_domain}:8080 "
+        f"--wallet {wallet}  object put --cid {cid} --timeout {PUT_TIMEOUT}s"
+    )
+    if expire_at is not None and expire_at > 0:
+        base_cmd += f" --expire-at {expire_at}"
+
+    relative_directory = os.path.relpath(directory)
+    for subdir, dirs, files in os.walk(relative_directory):
+
+        for filename in files:
```

Comment:
I would suggest to move this loop _body_ to separate function, like _push_file_ with `filename` or `filepath` argument

Reply:
> I would suggest to move this loop _body_ to separate function, like _push_file_ with `filename` or `filepath` argument

Fixed